### DISCUSSION TITLE
Tidak bisa buat Virtual Environment

### DISCUSSION THREAD

**Discussion by SKumulia** [teamalgoritma/community#37] [Q&A] [answered]
Anaconda Python sudah ter-intstall. Pada pengecekan version dengan command "python --version", hasil nya Python 3.9.7.

Saat akan membuat virtual environment dengan command "conda create --name da_rhea python3.9" tidak dapat diproses, karena "PackagesNotFoundError: The following packages are not available from current channels:  - python3.9"

Apakah ada yg terlewati oleh saya?

**Comment by onlyphantom** [comment, marked as answer]
Try this instead: 
```
conda create -n "da_rhea" python=3.9
```

This makes sure python3.9 is used to create the virtual environment. Your initial command makes it such that python 3.9 was interpreted to be a package. 

Good luck!

**Reply by onlyphantom** [reply]
A bit more information about the PackagesNotFound error message 😄 

`conda create -n “da_rhea” pandas`

this is interpreted as “create a virtual environment named `da_rhea` and install `pandas` package in it.

`conda create -n “da_rhea” python3.9`

this is interpreted  as “create a virtual environment named `da_rhea` and install `python3.9` package into it. Since it couldn’t find this package it returns the exception PackagesNotFound.

**Reply by SKumulia** [reply]
Thank you. It does work.

I was following the command and instruction from the recording videos from the first week. I think I missed the "=" sign. 

So what interpretation it made with the additional "=" sign?

**Reply by onlyphantom** [reply]
So in a simple way, you pass in parameters in one of three ways:
1. `awesometool -p ~/desktop/path`
2. `awesometool path='desktop/path'`
3. `awesometool ~desktop/path`

All of them does the same thing, but let me explain.

In (1), you use a shortflag called `-p` which refers to `path`. In (2), you explicitly use the name argument to say `path=____`. In (3) you're using positional argument. 

When you call `conda create --name da_rhea python3.9` the `python3.9` is taken as positional (case number 3), and is hence interpreted as a library you want to install. When you change it to:

```
conda create -n "da_rhea" python=3.9 pandas
```
Now you're using method (2) above, aka named parameter, to specify the version of python. Then you're using (3) positional to tell `conda` that **after your virtual environment is created** install pandas into it. 

How argument parsers work is a bit more than that, I have a video that goes into detail, line by line, how to implement one:
[Argument Parsers python tutorial](https://www.youtube.com/watch?v=DpvnLO6vw2I&list=PLXsFtK46HZxXIVE4tRjwMjwKFVaQSdT5W)

If the solution solves your problem, please "Accept it as answer :pray:". Good luck!
